src/core/distributions/phylogenetics/tree/birthdeath/OccurrenceBirthDeathProcess.cpp

```diff
@@ -0,0 +1,543 @@
+
+     // variable declarations and initialization
+     double lnProbTimes = computeLnProbabilityTimes();
+
+     return lnProbTimes;
+ }
+
+
+
+
+
+/**
+ * Compute the log probability of the current value under the current parameter values.
+ */
+double OccurrenceBirthDeathProcess::computeLnProbabilityTimes( void ) const
+{
+  updateVectorParameters();
+
+  // compute the log-likelihood : use ComputeLikelihoodsBackwardsLt (backward traversal of the tree) or ComputeLikelihoodsForwardsMt (forward traversal of the tree)
+  const RevBayesCore::Tree tree(*value);
+
+  std::vector<double> occAges;
+  if (occurrence_ages != NULL)
+  {
+      occAges = occurrence_ages->getValue();
+  }
+  else
+  {
+      occAges = std::vector<double>();
+  }
+
+  double logLikelihood = RevBayesCore::ComputeLnLikelihoodOBDP(start_age->getValue(), timeline, lambda, mu, psi, omega, homogeneous_rho, r, maxHiddenLin, cond, useMt, verbose, occAges, tree);
+
+  return logLikelihood;
+}
+
+
+
+
+
+/**
+ * return the index i so that t_{i-1} <= t < t_i
+ * where t_i is the instantaneous sampling time (i = 0,...,l)
+ * t_0 = 0.0
+ * t_l = Inf
+ */
+size_t OccurrenceBirthDeathProcess::findIndex(double t) const
+{
+    // Linear search for interval because std::lower_bound is not cooperating
+    if (timeline.size() == 2) // If timeline.size() were 1, we would have 0 break points and would be in constant-rate version
+    {
+      return(t < timeline[1] ? 0 : 1);
+    }
+    else
+    {
+      for (size_t i=0; i < timeline.size()-1; ++i)
+      {
+          if (t >= timeline[i] && t < timeline[i+1])
+          {
+              return(i);
+          }
+      }
+
+      return(timeline.size() - 1);
+    }
+}
+
+
+
+
+
+// calculate offset so we can set t_0 to time of most recent tip
+void OccurrenceBirthDeathProcess::getOffset(void) const
+{
+    offset = RbConstants::Double::max;
+    for (size_t i = 0; i < value->getNumberOfNodes(); i++)
+    {
+        const TopologyNode& n = value->getNode( i );
+
+        if ( n.getAge() < offset )
+        {
+            offset = n.getAge();
+        }
+    }
+}
+
+
+
+
+
+/**
+ * Here we ensure that we have a length-l vector of rates and events for all parameters and the timeline.
+ * In the case of homogeneous/constant-rate parameters, we fill the vector with this rate.
+ * In the case of homogenous/single events, we make all but the first event rates 0.
+ *
+ */
+void OccurrenceBirthDeathProcess::updateVectorParameters( void ) const
+{
+    // clean and get timeline
+    timeline.clear();
+    timeline = interval_times->getValue();
+
+    // Add t_0
+    getOffset();
+    timeline.insert(timeline.begin(),offset);
+
+    // clean all the sets
+    lambda.clear();
+    mu.clear();
+    psi.clear();
+    r.clear();
+    omega.clear();
+    psi_event.clear();
+
+    // Get vector of birth rates
+    if ( heterogeneous_lambda != NULL )
+    {
+      lambda = heterogeneous_lambda->getValue();
```

Comment:
make sure indentation is consistent within file. We usually use 4 spaces to indent.